platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -55,9 +55,11 @@ abstract class CrossTaskIncrementalJavaCompilationIntegrationTest extends Abstra
         """
         file("impl/build.gradle") << """
             def layout = project.layout
-            compileJava.doFirst {
-                options.compilerArgs << "--module-path" << classpath.join(File.pathSeparator)
-                classpath = layout.files()
+            compileJava {
+                options.compilerArgs.addAll(["--module-path", classpath.join(File.pathSeparator)])
```

Comment:
This is not the same as it resolves classpath earlier

Reply:
The problem is, that `ListProperty` is finalized and can't be modifed at the execution time. 
I think JavaCompile behavior stays the same even if we configure that earlier, but I need to double check it